contrib/macdeploy/detached-sig-create.sh

```diff
@@ -8,9 +8,11 @@ set -e
 
 ROOTDIR=dist
 BUNDLE="${ROOTDIR}/Bitcoin-Qt.app"
+BINARY="${BUNDLE}/Contents/MacOS/Bitcoin-Qt"
 SIGNAPPLE=signapple
 TEMPDIR=sign.temp
-OUT=signature-osx.tar.gz
+ARCH=$(file ${BINARY} | cut -d " " -f 4)
```

Comment:
When I test this PR it's grabbing `executable` rather than `x86_64` or `arm64`.

```suggestion
ARCH=$(file ${BINARY} | cut -d " " -f 5)
```

 With the above change I instead get:
```bash
Code signature created
Created signature-osx-arm64.tar.gz
....
Code signature created
Created signature-osx-x86_64.tar.gz
```

`file` output for either binary:
```bash
file dist/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt 
dist/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt: Mach-O 64-bit executable arm64
...
file dist/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt
dist/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt: Mach-O 64-bit executable x86_64
```

Reply:
Maybe we can use `awk` to get the last element?

`ARCH=$(file ${BINARY} | awk '{print $NF}')`